app/models/user.py

```diff
@@ -352,3 +355,19 @@ def __setattr__(self, name, value):
 @event.listens_for(User, 'init')
 def receive_init(target, args, kwargs):
     target.signup_at = datetime.now(pytz.utc)
+
+
+@event.listens_for(User, 'after_update')
```

Comment:
@SaptakS I think we mailing and notifications can work better as events occuring and listening in our app. This way we have separate module/class/file handeling different pre & post actions. This will also ensure that if any such action is done to models directly to helpers then the required pre/post process is also executed.
Also I think this binds an SQLAlchemy event to User model so may be we can create a separate helper handling different events and put this logic there?